account_analytic_parent/models/account_analytic_account.py

```diff
@@ -59,3 +59,25 @@ def name_get(self):
                 current = current.parent_id
             res.append((account.id, name))
         return res
+
+    @api.multi
+    @api.constrains('active')
+    def check_parent_active(self):
+        for account in self:
+            if (account.active and account.parent_id and
+                    account.parent_id not in self and
+                    not account.parent_id.active):
+                raise UserError(
+                    _('Please activate first parent account %s')
+                    % account.parent_id.display_name)
+
+    @api.multi
+    def archive_account(self):
+        for account in self:
+            account.active = False
+
+    @api.multi
+    def write(self, vals):
+        if 'active' in vals and not vals['active']:
+            self.mapped('child_ids').archive_account()
```

Comment:
```suggestion
            self.mapped('child_ids').write({'active': False})
```
and remove extra method